From 6b25abca848fe00a46536c61bd6770d219f965c8 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Jul 2026 11:08:41 +0200
Subject: [PATCH] tests: py: missing json output in meta.t with vlan mapping

JIRA: https://issues.redhat.com/browse/RHEL-190549
Upstream Status: nftables commit 5fec559727ffd2c6c8958748beab782096385758

commit 5fec559727ffd2c6c8958748beab782096385758
Author: Pablo Neira Ayuso <pablo@netfilter.org>
Date:   Mon Dec 11 12:54:35 2023 +0100

    tests: py: missing json output in meta.t with vlan mapping

    Fix this warning due to missing coverage:

     tests/py/any/meta.t.json.got: WARNING: line 2: Wrote JSON equivalent for rule meta mark set vlan id map { 1 : 0x00000001, 4095 : 0x00004095 }
     ERROR: did not find JSON equivalent for rule 'meta mark set vlan id map @map1

    Fixes: 8d3de823b622 ("evaluate: reset statement length context before evaluating statement")
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 tests/py/any/meta.t.json | 60 ++++++++++++++++++++++++++++++++++++++++
 1 file changed, 60 insertions(+)

diff --git a/tests/py/any/meta.t.json b/tests/py/any/meta.t.json
index 4734bbf..d50272d 100644
--- a/tests/py/any/meta.t.json
+++ b/tests/py/any/meta.t.json
@@ -2758,3 +2758,63 @@
         "accept": null
     }
 ]
+
+# meta mark set vlan id map { 1 : 0x00000001, 4095 : 0x00004095 }
+[
+    {
+        "mangle": {
+            "key": {
+                "meta": {
+                    "key": "mark"
+                }
+            },
+            "value": {
+                "map": {
+                    "data": {
+                        "set": [
+                            [
+                                1,
+                                1
+                            ],
+                            [
+                                4095,
+                                16533
+                            ]
+                        ]
+                    },
+                    "key": {
+                        "payload": {
+                            "field": "id",
+                            "protocol": "vlan"
+                        }
+                    }
+                }
+            }
+        }
+    }
+]
+
+# meta mark set vlan id map @map1
+[
+    {
+        "mangle": {
+            "key": {
+                "meta": {
+                    "key": "mark"
+                }
+            },
+            "value": {
+                "map": {
+                    "data": "@map1",
+                    "key": {
+                        "payload": {
+                            "field": "id",
+                            "protocol": "vlan"
+                        }
+                    }
+                }
+            }
+        }
+    }
+]
+
